pkg/apis/pipeline/v1/artifact_types.go

```diff
@@ -27,3 +47,72 @@ type Artifacts struct {
 	Inputs  []Artifact `json:"inputs,omitempty"`
 	Outputs []Artifact `json:"outputs,omitempty"`
 }
+
+func (a *Artifacts) Merge(another Artifacts) {
+	inputMap := make(map[string][]ArtifactValue)
+	var newInputs []Artifact
+
+	for _, v := range a.Inputs {
+		inputMap[v.Name] = v.Values
+	}
+
+	for _, v := range another.Inputs {
+		_, ok := inputMap[v.Name]
+		if !ok {
+			inputMap[v.Name] = []ArtifactValue{}
+		}
+		for _, vv := range v.Values {
+			exists := false
+			for _, av := range inputMap[v.Name] {
+				if cmp.Equal(vv, av) {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				inputMap[v.Name] = append(inputMap[v.Name], vv)
+			}
+		}
+	}
+
+	for k, v := range inputMap {
+		newInputs = append(newInputs, Artifact{
+			Name:   k,
+			Values: v,
+		})
+	}
+
+	outputMap := make(map[string][]ArtifactValue)
+	var newOutputs []Artifact
+	for _, v := range a.Outputs {
+		outputMap[v.Name] = v.Values
+	}
+
+	for _, v := range another.Outputs {
+		_, ok := outputMap[v.Name]
+		if !ok {
+			outputMap[v.Name] = []ArtifactValue{}
+		}
+		for _, vv := range v.Values {
+			exists := false
+			for _, av := range outputMap[v.Name] {
+				if cmp.Equal(vv, av) {
```

Comment:
Any reason to use `cmp.Equal` here ?

Reply:
vv, av are instances of ArtifactValue struct, this struct contains a map, we so cannot use `==` operator to compare them, that's why cmp.Equal is used here. 
```golang
type ArtifactValue struct {
	Digest map[Algorithm]string `json:"digest,omitempty"` // Algorithm-specific digests for verifying the content (e.g., SHA256)
	Uri    string               `json:"uri,omitempty"`    // Location where the artifact value can be retrieved
}
```